dbms/src/Storages/DeltaMerge/RowKeyFilter.h

```diff
@@ -43,38 +44,21 @@ inline Block cutBlock(Block && block, std::vector<std::pair<size_t, size_t>> off
         if (offset == 0 && limit == rows)
             return std::move(block);
 
-        if (offset == 0)
+        for (auto & col : block)
         {
-            size_t pop_size = rows - limit;
-            for (size_t i = 0; i < block.columns(); i++)
-            {
-                auto & column = block.getByPosition(i);
-                auto mutate_col = (*std::move(column.column)).mutate();
-                mutate_col->popBack(pop_size);
-                column.column = std::move(mutate_col);
-            }
-        }
-        else
-        {
-            for (size_t i = 0; i < block.columns(); i++)
-            {
-                auto & column = block.getByPosition(i);
-                auto new_column = column.column->cloneEmpty();
-                new_column->insertRangeFrom(*column.column, offset, limit);
-                column.column = std::move(new_column);
-            }
+            col.column = col.column->cut(offset, limit);
```

Comment:
when offset == 0，pop_back(xx) is faster than cut(xxx) because it avoids the overhead of memory copying

Reply:
reverted